app/controllers/books_controller.rb

```diff
@@ -1,14 +1,18 @@
 class BooksController < ApplicationController
-  before_action :set_book, only: %i[show edit update destroy]
+  before_action :set_book, only: %i[edit update destroy]
+  before_action :set_user
 
   def index
-    @books = Book.all
+    @user = set_user
+    @books = @user.books
   end
 
-  def show; end
+  def show
+    @book = Book.find(params[:id])
```

Comment:
So the idea is that any user can see any book (from current user or from any other user). But since any book 'dependent' on user, I changed it like this:
```
 def show
    @book = @user.books.find_by(id: params[:id])
 end
```